src/sys.zig

```diff
@@ -443,6 +443,90 @@ pub fn Maybe(comptime ReturnTypeT: type) type {
     return JSC.Node.Maybe(ReturnTypeT, Error);
 }
 
+/// Returns `true` if the two directories are on the same filesystem
+///
+/// This function does this test by trying to rename a file named from `dir_with_file_to_rename` -> `dir_to_rename_file_to`
+///
+/// The file must have the name `tmpname`
+pub fn testSameFileSystem(
+    dir_with_file_to_rename: std.fs.Dir,
+    dir_to_rename_file_to: std.fs.Dir,
+    tmpname: [:0]const u8,
+) bool {
+    // Make sure cachedir and cwd are in the same filesystem
+    std.posix.renameatZ(dir_with_file_to_rename.fd, tmpname, dir_to_rename_file_to.fd, tmpname) catch return false;
```

Comment:
None of these should be using `std.posix`

Reply:
and this doesn't test they're in the same filesystem. This tests it doesn't throw an error.

Reply:
and it shouldn't be in bun.sys unless it's reusable elsewhere